R/plot_forecast.R

```diff
@@ -138,54 +160,103 @@ plot_forecast <- function(forecast_data,
-        1-as.numeric(2*l)
-      })
-      
-      # for readability
-      if (all(c(.5, .8, .95) %in% intervals)){
-        intervals <- c(.5, .8, .95)
+      # only plot .95 interval if more than 5 models are selected
+      if (length(unique(models)) > 5){
+        intervals <- c(.95)
+      } else {
+        # generate intervals from lower bounds
+        intervals <- lapply(lower_bounds, function(l){
+          1-as.numeric(2*l)
+        })
+        
+        # for readability
+        if (all(c(.5, .8, .95) %in% intervals)){
+          intervals <- c(.5, .8, .95)
+        }
       }
     }
   }
   
+  if (!is.null(intervals) & length(unique(models)) > 5){
+    intervals <- c(.95)
+  }
+  
   # generate quantiles based on given intervals
   quantiles_to_plot <- unlist(lapply(intervals, function(interval){
     c(0.5 - as.numeric(interval)/2,
      0.5 + as.numeric(interval)/2)
   }))
   
   
-  # prediction interval shades
-  if (!is.null(intervals)){
-    
-    colourCount = length(quantiles_to_plot)/2+1
+  # set colors
+  if (fill_by_model){
+    if (length(unique(models)) <= 5){
+      color_families <- c("Blues", "Oranges", "Greens", "Purples", "Reds")
+      colourCount <- length(quantiles_to_plot)/2+1
+      model_colors <- purrr::map(
+        color_families[1:length(unique(models))],
+        function(color_family){
+          getPalette = colorRampPalette(RColorBrewer::brewer.pal(4, color_family))
+          getPalette(colourCount)
+          }
+        )
+      
+      ribbon_colors <- RColorBrewer::brewer.pal(4, "Greys")[1:3]
+      forecast_colors <- unlist(lapply(model_colors, tail, n = 1))
+      interval_colors <- unlist(lapply(model_colors, head, n = colourCount-1))
+    } else {
+      # interpolate color pallets to more than 5 colors
+      colourCount <- length(quantiles_to_plot)/2
+      modelCount <- length(unique(models))
+      # use 8 instead of 9 to avoid black and grey shades
+      getPalette = colorRampPalette(RColorBrewer::brewer.pal(8, "Set1"))
+      model_colors <- getPalette(modelCount)
+      
+      ribbon_colors <- RColorBrewer::brewer.pal(4, "Greys")[1:3]
+      forecast_colors <- unlist(lapply(model_colors, tail, n = 1))
+      # create 95% PI color
+      interval_colors <- unlist(lapply(model_colors, function(color){
+        colorspace::adjust_transparency(color, 0.3)}
+        ))
+    }
+  } else {
+    # only use blue
+    colourCount = max(length(quantiles_to_plot)/2+1, 2)
     getPalette = colorRampPalette(RColorBrewer::brewer.pal(4, "Blues"))
     blues = getPalette(colourCount)
-  } else {
-    blues <- RColorBrewer::brewer.pal(n=4, "Blues")
+    forecast_colors <- rep(tail(blues,1), length(unique(models)))
+    interval_colors <- rep(blues[1:(length(blues)-1)],
+                           length(unique(models)))
+    ribbon_colors <- blues[1:(length(blues)-1)]
   }
   
   # include truth from remote git hub repo by default
   # not using truth_as_of if we are loading truth from git hub repos
```

Comment:
Maybe we should throw a warning saying it's not being used if `truth_as_of` is provided in this case